pykafka/producer.py

```diff
@@ -387,9 +387,13 @@ def mark_as_delivered(message_batch):
                 for msg in mset.messages:
                     if (non_recoverable or msg.produce_attempt >= self._max_retries):
                         self._delivery_reports.put(msg, exc)
+                        log.debug("Message not delivered!! %r" % exc)
                     else:
                         msg.produce_attempt += 1
                         self._produce(msg)
+        else:
```

Comment:
It would probably make more sense to count the number of responses without errors (at line 351) and log that number after the loop. The current way you're doing it will only log if there were no errored messages.